src/WordPress/SubdomainChecker.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace CF\WordPress;
+
+// WordPressAPI::getDomainList could be use in hooks which doesn't 
+// rely on "GET /zones" request SubdomainChecker solves this problem
+// by changing WordPressAPI::domainName
+class SubdomainChecker
+{
+    private $originalDomain = '';
+    private $subDomain = '';
+
+    public function changeDomainName()
+    {
+        $config = new \CF\Integration\DefaultConfig('[]');
+        $logger = new \CF\Integration\DefaultLogger($config->getValue('debug'));
+        $dataStore = new \CF\WordPress\DataStore($logger);
+        $wordpressAPI = new \CF\WordPress\WordPressAPI($dataStore);
+        $wordpressIntegration = new \CF\Integration\DefaultIntegration($config, $wordpressAPI, $dataStore, $logger);
+        $clientAPIClient = new \CF\WordPress\WordPressClientAPI($wordpressIntegration);
```

Comment:
Hard coding your dependencies like this makes `changeDomainName()` hard to test because you can't mock any of the objects you instantiate with `new`.  This is why we use constructor based dependency injection.  

If this method is only used in `WordPressAPI::domainName` why not move it into that class?  

If you decide to keep this as a class, move the dependencies into the constructor so we can test them, otherwise just move it into `WordPressAPI`.  Either way we need tests for this method too.